libimage/runtime.go

```diff
@@ -316,19 +316,36 @@ func (r *Runtime) LookupImage(name string, options *LookupImageOptions) (*Image,
 		return img, name, err
 	}
 
-	return r.lookupImageInDigestsAndRepoTags(name, options)
+	return r.lookupImageInRepoTags(name, namedName, options)
 }
 
 // lookupImageInLocalStorage looks up the specified candidate for name in the
 // storage and checks whether it's matching the system context.
 func (r *Runtime) lookupImageInLocalStorage(name, candidate string, options *LookupImageOptions) (*Image, error) {
 	logrus.Debugf("Trying %q ...", candidate)
+
+	// First, try store.Image() which will properly work on IDs (but not on
+	// digests).
 	img, err := r.store.Image(candidate)
 	if err != nil && !errors.Is(err, storage.ErrImageUnknown) {
 		return nil, err
 	}
 	if img == nil {
-		return nil, nil
+		// Second, try parsing the candidate into a storage reference
+		// which will work with digests.  However, we must reparse the
+		// reference another time below since an ordinary image may
+		// have been referenced via its parent (manifest list) digest.
+		ref, err := storageTransport.Transport.ParseStoreReference(r.store, candidate)
```

Comment:
This accepts way more string values than necessary for digests, including things like `[foo]name:tag@digest@idPrefix`.

Instead, use
```go
referenceNamed = reference.TagNameOnly(referenceNamed) // might not actually be necessary
storageTransport.Transport.NewStoreReference(r.store, referenceNamed, nil)
```

As it happens, every single non-ID caller of `lookupImageInLocalStorage` already holds a `reference.Named` value, so we don’t need to parse _any_ string to get here, and that whole ambiguity risk can disappear.